File: learning/simple-linear-regression/simple_linear_regression.py
"""
Yulong Tan
11.14.17
"""

# Simple Linear Regression


# Importing the libraries
import numpy as np
import matplotlib.pyplot as plot
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature Scaling (Most of the time, a library will do this for us)
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

# Fitting a Simple Linear Regression to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting the test set results
y_prediction = regressor.predict(X_test)

# Visualizing the training set results
plot.scatter(X_train, y_train, color = 'red')
plot.plot(X_train, regressor.predict(X_train), color = 'blue')
plot.title('Salary vs Experience (Training Set)')
plot.xlabel('Years of Experience')
plot.ylabel('Salary')
plot.show()

# Visualizing the test set results
plot.scatter(X_test, y_test, color = 'red')
plot.plot(X_train, regressor.predict(X_train), color = 'blue')
plot.title('Salary vs Experience (Test Set)')
plot.xlabel('Years of Experience')
plot.ylabel('Salary')
plot.show()

File: learning/simple-linear-regression/simple_linear_regression.R
# Yulong Tan
# 11.14.17

# Simple Linear Regression

# Importing the dataset and libraries
library(ggplot2)
dataset = read.csv('Salary_Data.csv')

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set = scale(training_set)
# test_set = scale(test_set)

# Fitting Simple Linear Regression to the Training set
regressor <- lm(formula = Salary ~ YearsExperience,
                data = training_set)

# Predicting the test set results
y_prediction <- predict(regressor, newdata = test_set)

# Visualizing the training set results
ggplot() + 
  geom_point(mapping = aes(x = training_set$YearsExperience, y = training_set$Salary),
             color = 'red') +
  geom_line(mapping = aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary vs Experience (Training Set)') +
  xlab('Years of Experience') +
  ylab('Salary')

# Visualizing the test set results
ggplot() + 
  geom_point(mapping = aes(x = test_set$YearsExperience, y = test_set$Salary),
             color = 'red') +
  geom_line(mapping = aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary vs Experience (Test Set)') +
  xlab('Years of Experience') +
  ylab('Salary')